server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -668,14 +683,21 @@ UploadedMetadataResults writeMetadataInParallel(
                     clusterState,
                     indexRoutingTable,
                     listener,
-                    getClusterMetadataBasePath(
-                        blobStoreRepository,
-                        clusterState.getClusterName().value(),
-                        clusterState.metadata().clusterUUID()
-                    )
+                    clusterBasePath
                 )
             );
         });
+        if (indexRoutingTableDiff != null) {
+            uploadTasks.put(
+                InternalRemoteRoutingTableService.INDEX_ROUTING_DIFF_FILE_PREFIX + clusterState.getMetadata().clusterUUID(),
```

Comment:
Why is `clusterState.getMetadata().clusterUUID()` required here?

Reply:
Removed it, we don't need it.